x/protocolpool/keeper/keeper.go

```diff
@@ -67,3 +88,121 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	}
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
+
+func (k Keeper) ClaimFunds(ctx context.Context, recipient sdk.AccAddress) (amount sdk.Coin, err error) {
+	// get claimable funds from distribution info
+	amount, err = k.getClaimableFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	// distribute amount from feepool
+	err = k.DistributeFromFeePool(ctx, sdk.NewCoins(amount), recipient)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	return amount, nil
+}
+
+func (k Keeper) SubmitBudgetProposal(ctx context.Context, recipient sdk.AccAddress, totalBudget sdk.Coin, startTime, tranches, period int64) (types.Budget, error) {
+	budgetProposal := types.NewBudgetProposal(recipient, totalBudget, startTime, tranches, period)
```

Comment:
if this function takes the new budget proposal then we can avoid a bech32 encode here since we check it in the message server and already have the string version